packages/playwright-test/src/index.ts

```diff
@@ -361,46 +370,26 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
       await Promise.all(contexts.map(ctx => Promise.all(ctx.pages().map(screenshotPage))));
     };
 
-    const onWillCloseContext = async (context: BrowserContext) => {
-      // When reusing context, we get all previous contexts closed at the start of next test.
-      // Do not record empty traces and useless screenshots for them.
-      if (reusedContexts.has(context))
-        return;
-      await stopTracing(context.tracing, (context as any)[kStartedContextTearDown]);
-      if (screenshotMode === 'on' || screenshotMode === 'only-on-failure') {
-        // Capture screenshot for now. We'll know whether we have to preserve them
-        // after the test finishes.
-        await Promise.all(context.pages().map(screenshotPage));
-      }
-    };
-
-    const onWillCloseRequestContext =  async (context: APIRequestContext) => {
-      const tracing = (context as any)._tracing as Tracing;
-      await stopTracing(tracing, (context as any)[kStartedContextTearDown]);
-    };
-
     // 1. Setup instrumentation and process existing contexts.
+    const instrumentation = (playwright as any)._instrumentation as ClientInstrumentation;
+    instrumentation.addListener(csiListener);
     for (const browserType of [playwright.chromium, playwright.firefox, playwright.webkit]) {
-      (browserType as any)._onDidCreateContext = onDidCreateBrowserContext;
-      (browserType as any)._onWillCloseContext = onWillCloseContext;
       (browserType as any)._defaultContextOptions = _combinedContextOptions;
```

Comment:
Perhaps we can migrate this one as well, to something like `onWillCreateContext` or explicit `modifyNewContextOptions`?

Reply:
I wanted to follow up